### PR TITLE
Feat: Account Deletion Feature

### PR DESCRIPTION
**Pull request by abhishek310001**
## Description
_The goal of this PR is to add the feature of deleting an existing account.
Added page for accessing profile information as well as Accunt related actions._

## Additions
- _Added a Profile Settings page._

## Changes
- Modified `Sidebar Menu Item` to handle Sub Menus.
- Added a common layout to `main` that is used throughout the app

## Screenshots
Screenshot 2024-07-23 at 18 42 45
Screenshot 2024-07-25 at 13 38 40
Screenshot 2024-07-25 at 13 38 47
Screenshot 2024-07-23 at 18 46 41

## Videos

_Video demonstrating usage of Account Deletion feature with responsiveness._

https://github.com/user-attachments/assets/9290d13c-3a59-423c-a145-0826ed911751

## Database schema changes
_NA_

### Automated test cases added
- _DELETE /accounts/:accountId_
  -  should delete an account
  - should throw an error when deleting an account with an account ID that does not exist

### Manual test cases run
_For each manual test case, list the steps to test or reproduce the PR._
- _Checked manually using postman if account is deleting._
- _Checked if account is deleted or not after deleting account from frontend by looking into the db._
- _Since an existing component is modified, checked if didn't break functionality while maintaining the new one._

### CODE REVIEW

File: src/apps/frontend/services/account.service.ts [hunk 1]
```diff
@@ -15,4 +15,12 @@ export default class AccountService extends APIService {
       },
     });
   };
+
+  deleteAccount = async (): Promise<ApiResponse<void>> => {
+    const userAccessToken = JSON.parse(
```
Comment (bbbneo333) [inline src/apps/frontend/services/account.service.ts:20]: Think this should be a common logic defined which can be reused in our code. Ideally - This should be within the same code where the token is being saved in storage in the first place.

File: src/apps/frontend/types/account.ts [hunk 2]
```diff
@@ -1,19 +1,26 @@
...
     this.lastName = json.lastName as string;
+    this.phoneNumber = json.phoneNumber as PhoneNumber;
     this.username = json.username as string;
   }
 
   displayName(): string {
     return (`${this.firstName} ${this.lastName}`).trim();
   }
+
+  displayPhoneNumber(): string {
```
Comment (bbbneo333) [inline src/apps/frontend/types/account.ts:23]: Think this function should exist in `PhoneNumber`

**Review by bbbneo333 — CHANGES_REQUESTED** [review]
Overall, looks good! Few comments and one feedback on UI - "Delete Account" button should be in red.

File: src/apps/frontend/types/account.ts (hunk 2 above)
Comment (abhishek310001) [reply src/apps/frontend/types/account.ts]: modified existing PhoneNumber type to get this functionality.

File: src/apps/frontend/services/account.service.ts (hunk 1 above)
Comment (abhishek310001) [reply src/apps/frontend/services/account.service.ts]: Made a common logic for accessing, setting and removing access token from storage

**Comment by abhishek310001** [comment]
> Overall, looks good! Few comments and following feedback on UI:
> 
> Screenshot 2024-07-25 at 12 16 02
> 
> * "Delete Account" button can be in red
> * Not sure what's happening - But background color in text field should be same

"Delete Account" is now in red, updated the screenshots.
As for the Input Component I was using default disabled state seems like there was no styling for this state from the custom component causing the issue, edited it to make better.

**Review by bbbneo333 — APPROVED** [review]

**Pull request merged**